models/data_maps.go

```diff
@@ -298,3 +298,30 @@ func GetDataMapByGenesisHashAndChunkIdx(genesisHash string, chunkIdx int) ([]Dat
 
 	return dataMaps, err
 }
+
+func GetDataMap(genHash string, numChunks int) (dataMap []DataMap, vErr *validate.Errors) {
+
+	fileChunksCount := numChunks
+
+	if oyster_utils.BrokerMode != oyster_utils.TestModeNoTreasure {
+		fileChunksCount = oyster_utils.GetTotalFileChunkIncludingBuriedPearlsUsingNumChunks(numChunks)
+	}
+
+	currHash := genHash
+	for i := 0; i < fileChunksCount; i++ {
+		obfuscatedHash := oyster_utils.HashString(currHash, sha512.New384())
+		currAddr := string(oyster_utils.MakeAddress(obfuscatedHash))
+
+		dataMap := DataMap{
+			GenesisHash:    genHash,
+			ChunkIdx:       i,
+			Hash:           currHash,
+			ObfuscatedHash: obfuscatedHash,
+			Address:        currAddr,
+			Status:         Pending,
+		}
+		// Validate the data
+		vErr, _ = dataMap.Validate(nil)
+	}
```

Comment:
Maybe i'm reading this wrong but it looks like what would get returned from this method is just a single data_map (whatever the value of the last one is), not an array of data_maps.  I'd think you'd need to append all those data_maps to an array.

Reply:
I think this comment will resolve it: https://github.com/oysterprotocol/brokernode/pull/260#discussion_r186021976